src/models/gridOption.interface.ts

```diff
@@ -209,6 +209,9 @@ export interface GridOption<C extends BaseColumn = BaseColumn> {
   /** What is the minimum row buffer to use? */
   minRowBuffer: number;
 
+  /** Use a mixin function when applying defaults to passed in option and columns objects, rather than creating a new object, so as not to break references */
+  mixinDefaults: boolean;
```

Comment:
can you make it optional `?:`, I'm not entirely done with this interface. I had some issues with extending the interface in my libs, so I had to turn all options to optional in here so that I could extend properly in my libs. I'm still working on this though